### PR TITLE
Tools: python script cleanups (functionality unchanged)

### PR DESCRIPTION
**Pull request by BrandonPacewic**
Aligning the python scripts closer to the [PEP-8 style guide](https://peps.python.org/pep-0008/) & some other cleanups.

Functionality is completely unchanged, these suggestions are purely formatting and readability updates. Hopefully bringing some of this code closer to the incredible standards maintained by this team.

### CODE REVIEW

**Review by strega-nil-ms — APPROVED** [review]
This seems absolutely reasonable to me; I do like the type annotations. Otherwise, I don't personally have a lot of investment in following PEP-8 for our code, but also I don't mind following it either.

**Review by barcharcraz — APPROVED** [review]
Like nicole I don't really care about pep8 (and, in fact, I find it can sometimes have too many rules), but this seems OK to me.

Type annotations are always good, though I mostly use them just for the tooling support.

**Comment by BrandonPacewic** [comment]
> Otherwise, I don't personally have a lot of investment in following PEP-8 for our code, but also I don't mind following it either.

> Like nicole I don't really care about pep8 (and, in fact, I find it can sometimes have too many rules), but this seems OK to me.

I agree, following pep8 strictly isn't the best (especially with so many rules). Mainly just wanted to increase consistency.

File: tools/scripts/print_failures.py [hunk 1]
```diff
@@ -9,29 +9,29 @@
...
+            test_log = json.load(file)
+            for result in test_log["tests"]:
+                if not result["code"] in ["PASS", "UNSUPPORTED", "XFAIL"]:
+                    print(f"code: {result['code']}")
+                    # Ignore result["elapsed"].
+                    print(f"name: {result['name']}")
+                    # The JSON contains embedded CRLFs (which aren't affected by opening the file in text mode).
+                    # If we don't replace these CRLFs with LFs here, this script will appear to be okay in the console,
+                    # but redirecting it to a file will result in ugly double newlines.
+                    print(f"output: {result['output'].replace('\r\n', '\n')}")
```
Comment (StephanTLavavej) [inline tools/scripts/print_failures.py:36]: This breaks the script. How to repro:

1. Intentionally damage any STL test. Example:

```
C:\GitHub\STL\out\build\x64>git diff -U0
diff --git a/tests/std/tests/Dev11_0272959_make_signed/test.compile.pass.cpp b/tests/std/tests/Dev11_0272959_make_signed/test.compile.pass.cpp
index eff551f9..71a62cab 100644
--- a/tests/std/tests/Dev11_0272959_make_signed/test.compile.pass.cpp
+++ b/tests/std/tests/Dev11_0272959_make_signed/test.compile.pass.cpp
@@ -212 +212 @@ void example() {
-    STATIC_ASSERT(is_assignable_v<int&, short>);
+    STATIC_ASSERT(!is_assignable_v<int&, short>);
```

2. After building the STL, run that test by itself, and save a log file:

```
C:\GitHub\STL\out\build\x64>python tests\utils\stl-lit\stl-lit.py C:\GitHub\STL\tests\std\tests\Dev11_0272959_make_signed -o testing_x64.log
[...]
Testing Time: 3.73s
  Unsupported:  4
  Failed     : 27
```

3. Ask `print_failures.py` to print the log in a human-readable way. This now emits a `SyntaxError`:

```
C:\GitHub\STL\out\build\x64>python C:\GitHub\STL\tools\scripts\print_failures.py testing_x64.log
  File "C:\GitHub\STL\tools\scripts\print_failures.py", line 36
    print(f"output: {result['output'].replace('\r\n', '\n')}")
                                                             ^
SyntaxError: f-string expression part cannot include a backslash
```

For your convenience, I have uploaded this log file: [testing_x64.log](https://github.com/microsoft/STL/files/9642057/testing_x64.log)

**Review by StephanTLavavej — CHANGES_REQUESTED** [review]

**Comment by StephanTLavavej** [comment]
Thanks for this cleanup! I wanted to use f-strings when I wrote `print_failures.py` but didn't know how to deal with the nested quotes (since I'm a Python novice), so I'm glad you've shown us how to. There's an issue with the CRLF replacement - I don't know if that line needs to be reverted, or if it can be adjusted to keep the f-string improvement.

**Comment by BrandonPacewic** [comment]
My mistake for not double checking the backslash problem with f-strings, I have reverted that change for now. The other option would be something like this:

```python3
my_string = f"This is a backslash: {chr(92)}"
```

92 being the ASCII code for a backslash `\`. With the final product being something like this:

```python3
print(f"output: {result['output'].replace(f'{chr(92)}r{chr(92)}n', f'{chr(92)}n')}")
```

Definitely not the best solution but seems to be necessary to get away from `.format()` currently.

File: tools/scripts/print_failures.py (hunk 1 above)
Comment (BrandonPacewic) [reply tools/scripts/print_failures.py]: Syntax error now resolved; line change reverted.

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
:white_check_mark: No C++, so no modules impact! :smile_cat:

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
Thanks for improving these scripts - and congratulations on your first microsoft/STL commit! :tada: :smile_cat: :snake:

**Comment by BrandonPacewic** [comment]
> Thanks for improving these scripts - and congratulations on your first microsoft/STL commit! 🎉 😸 🐍

 🎉🚀🎉

**Pull request merged**